### PR TITLE
Add environment variable to override plugins dir

### PR DESCRIPTION
**Pull request by t1m0thyj**
This is the Imperative part of the enhancement requested in https://github.com/zowe/zowe-cli/issues/1483

### CODE REVIEW

File: packages/imperative/src/doc/IImperativeEnvironmentalVariableSettings.ts [hunk 1]
```diff
@@ -47,4 +47,10 @@ export interface IImperativeEnvironmentalVariableSettings {
      * @type {IImperativeEnvironmentalVariableSetting}
      */
     maskOutput?: IImperativeEnvironmentalVariableSetting;
+
+    /**
+     * The directory where CLI plugins are installed.
+     * Default is `${cliHome}/plugins`.
+     */
+    pluginsDir?: IImperativeEnvironmentalVariableSetting;
```
Comment (zFernand0) [inline packages/imperative/src/doc/IImperativeEnvironmentalVariableSettings.ts:55]: Thoughts on keep using annotations like `@type` ?

It doesn't affect the generated `typedoc`, Examples:
- with types defined: [IImperativeEnvironmentalVariableSettings](https://docs.zowe.org/stable/typedoc/interfaces/_zowe_imperative.iimperativeenvironmentalvariablesettings.html)
- without types defined: [IImperativeAuthGroupConfig](https://docs.zowe.org/stable/typedoc/interfaces/_zowe_imperative.iimperativeauthgroupconfig.html)

It appears in VSCode suggestions
image

However, the type already says it's a string, and you already know the parent class/interface by hovering over the object, so `@memberOf` could also be removed.
image

**Review by zFernand0 — COMMENTED** [review]
🥳 changes LGTM!
Left a question, but it's unrelated to the functionality 😋

**Review by KevinLoesch1 — APPROVED** [review]

File: packages/imperative/src/doc/IImperativeEnvironmentalVariableSettings.ts (hunk 1 above)
Comment (gejohnston) [reply packages/imperative/src/doc/IImperativeEnvironmentalVariableSettings.ts:55]: I have never been a fan of our (often redundant) typedoc annotations. I would not suggest that we make a concentrated effort to remove them everywhere. However, if we choose to not place them in new source files, and remove them when convenient in a file that we are already working on, we could remove clutter that distracts someone who is reading the code.

**Review by gejohnston — APPROVED** [review]

**Comment by zFernand0** [comment]
this PR will be combined with #879

**Pull request merged**